### PR TITLE
8254998: C2: assert(!n->as_Loop()->is_transformed_long_loop()) failure with -XX:StressLongCountedLoop=1

### PR DESCRIPTION
**Pull request by rwestrel**
Transformation of a long counted loop into a loop nest with a counted
loop inner loop happens in 2 passes of loop opts: first the loop nest
is created and then the inner loop is transformed into a counted
loop. The assert fires because the second step doesn't have a chance
to run given the first step was executed as the last loop opts pass
before running out of allowed loop opts passes. The fix I propose for
this corner case is to relax the assert to account for this.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux x64 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- |
| Build | ✔️ (5/5 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) |
| Test (tier1) | ✔️ (9/9 passed) | ✔️ (9/9 passed) | ✔️ (9/9 passed) |

### Issue
 * [JDK-8254998](https://bugs.openjdk.java.net/browse/JDK-8254998): C2: assert(!n->as_Loop()->is_transformed_long_loop()) failure with -XX:StressLongCountedLoop=1

### Reviewers
 * [Vladimir Ivanov](https://openjdk.java.net/census#vlivanov) (@iwanowww - **Reviewer**) ⚠️ Review applies to 3e6d02699fa831c416c3012e712605a153f225c1
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to 3e6d02699fa831c416c3012e712605a153f225c1
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/735/head:pull/735`
`$ git checkout pull/735`

### CODE REVIEW

**Review by iwanowww — APPROVED** [review]
Looks good.

**Review by vnkozlov — APPROVED** [review]
Good.

File: test/hotspot/jtreg/compiler/longcountedloops/TestTooManyLoopOpts.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug JDK-8254998
```
Comment (TobiHartmann) [inline test/hotspot/jtreg/compiler/longcountedloops/TestTooManyLoopOpts.java:26]: I didn't know that the bug ID format with "JDK-" is accepted as well but I would suggest to remove it for consistency with other tests.

**Review by TobiHartmann — APPROVED** [review]
Looks good to me.

File: test/hotspot/jtreg/compiler/longcountedloops/TestTooManyLoopOpts.java (hunk 1 above)
Comment (rwestrel) [reply test/hotspot/jtreg/compiler/longcountedloops/TestTooManyLoopOpts.java]: I didn't include "JDK-" on purpose. I've just fixed it.

**Review by TobiHartmann — APPROVED** [review]

**Comment by rwestrel** [comment]
@iwanowww @vnkozlov @TobiHartmann thanks for the review.

**Pull request closed without merging**